src/NumericalAlgorithms/LinearOperators/IndefiniteIntegral.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/LinearOperators/IndefiniteIntegral.hpp"
+
+#include <array>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/SliceIterator.hpp"
+#include "DataStructures/StripeIterator.hpp"
+#include "Domain/Mesh.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Blas.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+template <size_t Dim>
+void indefinite_integral_impl(const gsl::not_null<DataVector*> integral,
+                              const DataVector& integrand,
+                              const Mesh<Dim>& mesh,
+                              const size_t dim) noexcept {
+  if (UNLIKELY(integral->size() != integrand.size())) {
+    *integral = DataVector(integrand.size());
+  }
+  const Mesh<1> mesh_1d{mesh.extents(dim), gsl::at(mesh.basis(), dim),
+                        gsl::at(mesh.quadrature(), dim)};
+  const size_t num_pts = mesh_1d.number_of_grid_points();
+  const Matrix& indefenite_integration_matrix =
+      Spectral::integration_matrix(mesh_1d);
+  for (StripeIterator stripe_it(mesh.extents(), dim); stripe_it; ++stripe_it) {
+    dgemv_('N', num_pts, num_pts, 1.0, indefenite_integration_matrix.data(),
+           num_pts, integrand.data() + stripe_it.offset(),  // NOLINT
+           stripe_it.stride(), 0.0,
+           integral->data() + stripe_it.offset(),  // NOLINT
+           stripe_it.stride());
+  }
+}
+}  // namespace
+
+template <size_t Dim>
+void indefinite_integral(const gsl::not_null<DataVector*> integral,
+                         const DataVector& integrand,
+                         const Mesh<Dim>& mesh) noexcept {
+  if (UNLIKELY(integral->size() != integrand.size())) {
+    *integral = DataVector(integrand.size());
+  }
+  if (Dim == 1) {
+    indefinite_integral_impl(integral, integrand, mesh, 0);
+  } else if (Dim == 2) {
+    DataVector temp(integral->size());
+    indefinite_integral_impl(&temp, integrand, mesh, 0);
```

Comment:
It seems as though you are doing $g(x,y,z) = \int_0^x dx^\prime \int_0^y dy^\prime \int_0^z dz^\prime f(x^\prime, y^\prime, z^\prime)$, which is pretty concerning. This would be a path integral, but along a specific grid-based path. Note that this is a path-dependent operation whenever the integrand is not separable. Do you have a specific use-case for this so I can understand the strategy?

Reply:
Alright, for the limiter case you're looking at, where you look at integrals over sub-elements described by regions with collocation points on there corners, I think this is still a bit confusing, but it's seeming more alright.

For the boundary conditions, I think it is important that all of the 1-D integration steps use boundary condition 0, and that a single `double` value get added to every single collocation point as the only boundary condition. So, my main complaint at this point is just that the process should be well-documented wherever this function gets relocated to :smile: .

As a footnote for future discussion, I'm also unsure about how much sense these procedures make when the metric is not the kronecker delta.

Reply:
Alright, I will remove the multi-dimensional integral from this PR and clean it up in the FV subcell code. I will still document what we've discussed so that in the future if someone needs to extend it they will have an easier time :)